### PR TITLE
REG-69: rename endpoints for clarity and REST standards

### PR DESCRIPTION
**Pull request by sleidig**
suggesting to clean up the api endpoint urls and rename them more following the RESTful API patterns (see e.g. https://restfulapi.net/http-methods/)

### CODE REVIEW

File: api/GovStack_Registration_BB_API.json [hunk 1]
```diff
@@ -60,14 +58,12 @@
...
-				"tags": [
-					"Applicant"
-				],
+				"tags": ["Available Services"],
 				"summary": "returns detailed information about a service",
 				"description": "Pass in the ID of the service and it will return all information about that service\n",
 				"operationId": "getService",
 				"parameters": [{
-					"$ref": "#/components/parameters/serviceIdParam"
+					"$ref": "#/components/parameters/serviceKeyParam"
```
Comment (sleidig) [inline api/GovStack_Registration_BB_API.json:66]: let's make naming more consistent - maybe best use `serviceId`, which is more in line with the other parameters' naming?

File: api/GovStack_Registration_BB_API.json [hunk 2]
```diff
@@ -435,156 +430,65 @@
 			}
 		},
 
-		"/service/operatorFileList": {
```
Comment (sleidig) [inline api/GovStack_Registration_BB_API.json:438]: both operatorFileList and applicantFileList are actually doing the same (maybe with different access permissions). Similar to /tasks I suggest to just have a single endpoint getting existing applications (--> `/applications`)

File: api/GovStack_Registration_BB_API.json [hunk 3]
```diff
@@ -153,17 +149,17 @@
 			}
 		},
 
-		"/service/{serviceKey}/startWithForm": {
+		"/services/{serviceKey}/applications": {
```
Comment (sleidig) [inline api/GovStack_Registration_BB_API.json:152]: if we are following the REST standards (which are widely used in the industry), URLs should reflect "resources" not actions and there can be "sub-resources". Here e.g. applications belonging to a specific service

File: api/GovStack_Registration_BB_API.json [hunk 4]
```diff
@@ -754,18 +658,6 @@
 					}
 				}
 			},
-			"service": {
```
Comment (sleidig) [inline api/GovStack_Registration_BB_API.json:757]: seemed redundant to have service and serviceDefinition schemas separately?

File: api/GovStack_Registration_BB_API.json [hunk 5]
```diff
@@ -267,15 +264,15 @@
...
+				"tags": ["Applicant", "Operator"],
+				"summary": "Retrieve all application files",
+				"description": "Returns a list of application files the user has permission to access (i.e. either only the applicant's own applications or all applications for an operator to process)",
 				"operationId": "getApplicantFileList",
 				"parameters": [{
-						"$ref": "#/components/parameters/Information-Mediator-Client"
-					},
+					"$ref": "#/components/parameters/Information-Mediator-Client"
+				},
 					{
```
Comment (sleidig) [inline api/GovStack_Registration_BB_API.json:276]: Are the generic query parameters for sorting + pagination (e.g. firstResult, maxResult, ...) important enough for a BB spec at the higher level? Or would it be easier to read if we remove these and only specify things related to functional aspects of the BB (+ maybe add a general para in the GitBook that these querying details should be supported)?

File: api/GovStack_Registration_BB_API.json (hunk 1 above)
Comment (ingmarvali) [reply api/GovStack_Registration_BB_API.json:66]: Fine with me

File: api/GovStack_Registration_BB_API.json (hunk 2 above)
Comment (ingmarvali) [reply api/GovStack_Registration_BB_API.json:438]: I like the idea. You are free to optimize.

File: api/GovStack_Registration_BB_API.json (hunk 4 above)
Comment (ingmarvali) [reply api/GovStack_Registration_BB_API.json:757]: sounds good!

File: api/GovStack_Registration_BB_API.json (hunk 5 above)
Comment (ingmarvali) [reply api/GovStack_Registration_BB_API.json:276]: Good idea, thease are problably not relevant in the large view.

**Pull request merged**